File: 0x-bonus/yegge/sum_integers/sum_integers.rb
# def sum_integers(filename)
#   sum = 0
#   file = File.new filename
#   file.each_line { |line| sum += line.to_i }
#   sum
# end

def sum_integers(filename)
  File.readlines(filename).map(&:to_i).inject(&:+) # Symbol to proc
  # File.readlines(filename).map { |n| n.to_i }.inject { |acc, n| acc + n }
end

puts sum_integers('integers.txt')

# require 'pry'
# puts "you still have pry in your code"
# binding.pry

File: 0x-bonus/yegge/sum_integers/sumIntegers.js
const fs = require('fs'); // File System code from Node std library.

const sumIntegers = function (filename) {
  // TODO: Try this with fs.readFile -- the asynchronous version.
  const file = fs.readFileSync(filename, 'utf-8');
  return file.split('\n').reduce(function (acc, n) {
    return Number(acc) + Number(n);
  });

  // file.split('\n').forEach(function (line) {
  //   const n = Number(line);
  //   sum += n;
  // });
  // return sum;
};

console.log( sumIntegers('./integers.txt') );
